src/parse/parser.rs

```diff
@@ -0,0 +1,522 @@
+
+    }
+
+    // Parses the tokens for the column vector subtree
+    fn parse_create_column_vec(&mut self) -> Result<Vec<ColumnInfo>, ParseError> {
+        // Convention: Every method must use bump to
+        // put the lexer to the position of the token the method needs
+        self.bump();
+        let mut colsvec = Vec::<ColumnInfo>::new();
+
+        // fill the vector with content until ParenCl is the curr token
+        while match self.expect_token(&[Token::ParenCl]) {
+            Ok(&Token::ParenCl) => false,
+            _ => true,
+        } {
+            // parsing the content for a single ColumnInfo
+            colsvec.push(try!(self.expect_column_info()));
+            self.bump();
+            // Check if there is a Comma seperating two columns or a ParenCl ending the vectorparsing
+            match try!(self.expect_token(&[Token::Comma, Token::ParenCl])) {
+                &Token::Comma => {self.bump();() },
+                _ => (),
+            };
+        }
+        Ok(colsvec)
+    }
+
+
+
+
+
+
+    // Parses tokens for alter statement
+    fn parse_alt_stmt(&mut self) -> Result<AltStmt, ParseError> {
+        self.bump();
+
+        match try!(self.expect_keyword(&[Keyword::Table])) {
+            Keyword::Table => return Ok(AltStmt::Table(try!(self.parse_alter_table_stmt()))),
+
+            // Unknown parsing error
+            _=> return Err(ParseError::UnknownError),
+        };
+    }
+
+
+    // Parses table to modify and subsequent operations
+    fn parse_alter_table_stmt(&mut self) -> Result<AlterTableStmt, ParseError> {
+        self.bump();
+
+        let alt_table_stmt = AlterTableStmt { tid: try!(self.expect_word()), op: try!(self.parse_alter_op()) };
+        Ok(alt_table_stmt)
+
+    }
+
+    // Parses operations applied on selected table including tablename and datatype if necessary
+    fn parse_alter_op(&mut self) -> Result<AlterOp, ParseError> {
+        self.bump();
+        match try!(self.expect_keyword(&[Keyword::Add, Keyword::Drop, Keyword::Modify])) {
+            Keyword::Add => return { self.bump(); Ok(AlterOp::Add(try!(self.expect_column_info()))) },
+            Keyword::Drop => return { self.bump(); Ok(AlterOp::Drop(try!(self.expect_word()))) },
+            Keyword::Modify => return { self.bump(); Ok(AlterOp::Modify(try!(self.expect_column_info()))) },
+            _ => return Err(ParseError::UnknownError),
+        }
+    }
+
+    // Parses the tokens for drop statement
+    fn parse_drop_stmt(&mut self) -> Result<DropStmt, ParseError> {
+        self.bump();
+
+        match try!(self.expect_keyword(&[Keyword::Table,Keyword::Database])) {
+            Keyword::Table => return { self.bump();
+                Ok(DropStmt::Table(try!(self.expect_word())))},
+                Keyword::Database => return { self.bump();
+                    Ok(DropStmt::Database(try!(self.expect_word()))) },
+                    _ => return Err(ParseError::UnknownError),
+                };
+            }
```

Comment:
here you could do the same as i explained above to make it easier to read length is fine though.